### DISCUSSION TITLE
Dynamic registries

### DISCUSSION THREAD

**Discussion by AndersOstmanNovogit** [gopeg/issues-barriers#51] [General]
One issue reported in D2.2 geo-COVID Watch use case is the following. "The lack of clearly defined concepts pertaining to the COVID pandemic have been a serious hindrance in the aggregation of available data across geographic boundaries, as required to better understand the dynamics of this process." 

The following solution is proposed. "The creation of the GO-PEG registry, where above mentioned definitions are made publicly available through URIs, provides a good solution to this problem as these are therein structured in accordance with the requirements of the Re3gistry Software, including relevant relations between concepts." 

This may be an appropriate solution in the short term, but the problem also indicate a more general problem, namely how to deal with concepts where definitions are changing over time. The situation may appear when data from additional countries are to be included (where slightly modified concepts has been used) or when concepts are redefined for other reasons. This problem is not new. As an example , DYNTAXA (https://www.slu.se/dyntaxa/) is a database of Swedish species (names and relationships) which has been in use since late 90's. This database is designed to accommodate changes in definitions etc, hence the name (dyntaxa = dynamic taxonomy). It would surprise me if this is the oldest implementation of such a solution, but it is anyway one that I am aware of.

The question then is, to what extent are the INSPIRE registries prepared for dynamic changes of the entries? What happens if the meaning of a concept change over time? Is this something we should recommend for future developments?

**Comment by sMorrone** [comment]
The Re3gistry software, which underpins the GO-PEG code list registry, enables consistent management and versioning of reference codes. For example, when the definition of a concept is updated, the associated page is automatically updated with a link to a page exposing the previous definition. This means that the same (persistent) URI allows access to all versions of a concept (version history) - see Figure below illustrating the management of reference codes in the Re3gistry 
![image](https://user-images.githubusercontent.com/13329248/218042684-8da9e68c-ed56-4e17-8d7f-982cacf51702.png)

**Reply by AndersOstmanNovogit** [reply]
OK, that's excellent. The issue is then closed.
